tests/components/philips_js/test_config_flow.py

```diff
@@ -65,6 +66,41 @@ async def test_form(hass: HomeAssistant, mock_setup_entry) -> None:
     assert len(mock_setup_entry.mock_calls) == 1
 
 
+async def test_reauth(
+    hass: HomeAssistant, mock_setup_entry, mock_config_entry, mock_tv
+) -> None:
+    """Test we get the form."""
+
+    mock_tv.system = MOCK_SYSTEM | {"model": "changed"}
+
+    assert await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    assert len(mock_setup_entry.mock_calls) == 1
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={
+            "source": config_entries.SOURCE_REAUTH,
+            "unique_id": mock_config_entry.unique_id,
+            "entry_id": mock_config_entry.entry_id,
+        },
+        data=mock_config_entry.data,
+    )
+
+    assert result["type"] == "form"
```

Comment:
You can use `FlowResultType` constant here, and I always recommend that FORM checks are associated with a check on `step_id`:
```suggestion
    assert result["type"] == FlowResultType.FORM
    assert result["step_id"] == "user"
```